frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -109,7 +105,54 @@ scopeResolveModuleStmt(Context* context, ID id) {
   return QUERY_END(result);
 }
 
+static void updateTypeForModuleLevelSplitInit(Context* context, ID id,
+                                   ResolvedExpression& lhs,
+                                   const ResolvedExpression& rhs,
+                                   std::set<ID>& alreadyUpdated) {
+  const QualifiedType lhsType = lhs.type();
+  const QualifiedType rhsType = rhs.type();
+
+  // check to see if it is generic/unknown
+  // (otherwise we do not need to infer anything)
+  auto g = Type::MAYBE_GENERIC;
+  if (lhsType.isUnknownKindOrType()) {
+     // includes nullptr type, UnknownType, and param with unknown value
+     g = Type::GENERIC;
+  } else {
+    CHPL_ASSERT(lhsType.type()); // should not be nullptr b/c of isUnknownKindOrType
+    g = getTypeGenericity(context, lhsType.type());
+  }
```

Comment:
I could but I'm not sure this is worth extracting. I went to do so and realized it can be written as simply as:
```
if (!(lhsType.isUnknownKindOrType() ||
      getTypeGenericity(context, lhsType.type()) == Type::GENERIC)) {
  return;
}
```
It's so close to equivalent to `lhsType.isGenericOrUnknown`, but that would return true for `MAYBE_GENERIC` and `GENERIC_WITH_DEFAULTS` as well. And I don't think this is a common enough case to make a method on `QualifiedType` for, which is where this would most belong imo. Thoughts?